src/roman_datamodels/datamodels.py

```diff
@@ -179,6 +180,12 @@ def __setattr__(self, attr, value):
     def __getattr__(self, attr):
         return getattr(self._instance, attr)
 
+    def __setitem__(self, key, value):
```

Comment:
What happens if the user sets a key from the schema using the bracket operator?  Does this bypass validation?

Also, can the new attribute be set with dot notation after it's been created?

Reply:
I'll check the first, probably it does, so I should see if the attribute exists first. Yes, it should be modifiable by dot notation, but also a good thing to test.